### PR TITLE
GH-40954: [CI] Fix use of obsolete docker-compose command on Github Actions

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

The `docker-compose` utility is progressively being removed from GHA-provided runners:
https://github.com/actions/runner-images/issues/9557

### What changes are included in this PR?

Use `docker` client CLI directly instead of `docker-compose` where possible.

### Are these changes tested?

Yes, this should fix the sporadic CI failures because of the above migration.

### Are there any user-facing changes?

No, except additional optional env var `ARCHERY_DEBUG`.
* GitHub Issue: #40954

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit -g nightly

**Comment by pitrou** [comment]
@github-actions crossbow submit -g nightly

**Comment by pitrou** [comment]
@github-actions crossbow submit -g cpp

**Comment by pitrou** [comment]
@github-actions crossbow submit -g linux

**Comment by pitrou** [comment]
@github-actions crossbow *cuda*

**Comment by pitrou** [comment]
@github-actions crossbow submit *cuda*

**Comment by pitrou** [comment]
@github-actions crossbow submit *wheel*312*

File: dev/archery/archery/docker/cli.py [hunk 1]
```diff
@@ -46,10 +46,19 @@ def _execute(self, *args, **kwargs):
               callback=validate_arrow_sources,
               help="Specify Arrow source directory.")
 @click.option('--dry-run/--execute', default=False,
-              help="Display the docker-compose commands instead of executing "
-                   "them.")
+              help="Display the docker commands instead of executing them.")
+@click.option('--using-docker-cli', default=False, is_flag=True,
```
Comment (assignUser) [inline dev/archery/archery/docker/cli.py:50]: With docker-compose being deprecated, shouldn't we default to use docker cli? That way using the deprecated, potentially outdated tool requires an explicit opt-in.

Comment (pitrou) [reply dev/archery/archery/docker/cli.py:50]: We can probably default at some point, but I don't want to disrupt things too much here.

**Comment by pitrou** [comment]
@raulcd @assignUser @kou Any further comments here? Otherwise it would be nice to merge this quickly.

**Review by assignUser — APPROVED** [review]

**Pull request merged**